fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1518,8 +1518,9 @@ public static IndexState writeOnlyIfTooManyRecordsForRebuild(long recordCount, b
     @Nonnull
     private CompletableFuture<Boolean> checkVersion(@Nonnull RecordMetaDataProto.DataStoreInfo storeHeader, @Nullable UserVersionChecker userVersionChecker) {
         RecordMetaDataProto.DataStoreInfo.Builder info = storeHeader.toBuilder();
-        final int oldMetaDataVersion = info.hasMetaDataversion() ? info.getMetaDataversion() : -1;
-        final int oldUserVersion = info.hasUserVersion() ? info.getUserVersion() : -1;
+        final boolean newStore = info.getFormatVersion() == 0;
+        final int oldMetaDataVersion = newStore ? -1 : info.getMetaDataversion();
+        final int oldUserVersion = newStore ? -1 : info.getUserVersion();
```

Comment:
I suppose an alternative would be to make a static `DataStoreInfo` message for new stores with `metaDataversion` and `user_version` both set to -1, and the format version set to 0. I'm not sure if that removes some special cases or not.